packages/gestalt/src/Masonry.tsx

```diff
@@ -180,6 +186,32 @@ export default class Masonry<T> extends ReactComponent<Props<T>, State<T>> {
       props.measurementStore || Masonry.createMeasurementStore();
 
     this.positionStore = props.positionStore || Masonry.createMeasurementStore();
+    this.idxToDataMap = {};
```

Comment:
Idealy yes, the items go in order, although since the idx is set to the filtered `itemsToRender` it is not exactly 1-1, if for some reason there is an item on the array that doesn't have a measure but the others do then it is not going to match, if we don't want to use this map we could filter before getting the position index from the items array.

Reply:
I removed it for now but we can discuss if this is really an issue.